integration-test/src/main/java/com/sequenceiq/it/util/cleanup/CleanupUtil.java

```diff
@@ -172,21 +191,213 @@ public void cleanupCredentials() {
+                    for (int i = 0; i < resources.length(); i++) {
+                        String resource = resources.getString(i);
+                        resourceNames.add(resource);
+                        LOG.info("Get '{}' JSON array '{}' element from resource file with: '{}'.", resourceNameType, i, resource);
+                    }
+                } catch (JSONException e) {
+                    String resource = jsonObject.getString(resourceNameType);
+                    resourceNames.add(resource);
+                    LOG.info("Get '{}' JSON object from resource file with: '{}'.", resourceNameType, resource);
+                }
+            } else {
+                LOG.error("Cannot find '{}' in resource file '{}'.", resourceNameType, filePath.getFileName());
+            }
+            return resourceNames;
+        } catch (JSONException e) {
+            LOG.warn("Cannot get '{}' key, because of: {}", resourceNameType, e.getMessage(), e);
+            return resourceNames;
+        } catch (FileNotFoundException e) {
+            LOG.warn("'{}' file not found, because of: {}", filePath, e.getMessage(), e);
+            return resourceNames;
+        } catch (IOException e) {
+            LOG.warn("Reading '{}' file throws exception: {}", filePath, e.getMessage(), e);
+            return resourceNames;
+        }
+    }
+
+    private void deleteResources(List<String> foundResources, String resourceNameType) {
+        List<Path> fileList = new ArrayList<>();
+        List<String> deletedCredentials = new ArrayList<>();
+        AtomicBoolean e2eCleanupFailed = new AtomicBoolean(false);
+
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(Paths.get(outputDirectory))) {
+            for (Path path : stream) {
+                String fileName = path.getFileName().toString();
+                if (fileName.startsWith("resource_names") && fileName.endsWith(".json")) {
+                    LOG.info("Found resource file: '{}' is going to be added to resource files' list", path.getFileName().toAbsolutePath().normalize());
+                    fileList.add(path);
+                }
+            }
+        } catch (Exception e) {
+            LOG.error("Cannot find resource file at path: '{}', because of: {}", Paths.get(outputDirectory).toAbsolutePath().normalize(), e.getMessage(), e);
+            throw new RuntimeException(String.format("Cannot find resource file at path: '%s', because of: %s",
+                    Paths.get(outputDirectory).toAbsolutePath().normalize(), e.getMessage()));
+        }
+        fileList.forEach(filePath -> {
+            LOG.info("Processing resource file: '{}'", filePath.getFileName());
+            List<String> resourcesName = Optional.ofNullable(getResourcesFromFile(resourceNameType, filePath))
+                    .orElse(List.of(""));
+            resourcesName.forEach(resourceName -> {
+                if (foundResources.stream()
```

Comment:
Cool, functional, but I guess `foundResource.contains(resourceName)` will be enough instead of `foundResources.stream().anyMatch(resourceName::equals)`.